### PR TITLE
STM32F723 support

### PR DESCRIPTION
**Pull request by aurel32**
This patch series adds support for the STM32F723 SoC, including basic support for the STM32F723E-DISCO board.

### CODE REVIEW

File: boards/arm/stm32f723e_disco/Kconfig.board [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Kconfig - STM32F746G Discovery board configuration
```
Comment (ydamigos) [inline boards/arm/stm32f723e_disco/Kconfig.board]: STM32F723E?

File: boards/arm/stm32f723e_disco/pinmux.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <pinmux.h>
+#include <sys_io.h>
+
+#include <pinmux/stm32/pinmux_stm32.h>
+
+/* pin assignments for STM32F746G-DISCO board */
```
Comment (ydamigos) [inline boards/arm/stm32f723e_disco/pinmux.c]: STM32F723E Discovery?

File: boards/arm/stm32f723e_disco/pinmux.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+static const struct pin_config pinconf[] = {
+#ifdef CONFIG_UART_STM32_PORT_2
+	{STM32_PIN_PA2, STM32F7_PINMUX_FUNC_PA2_USART2_TX},
+	{STM32_PIN_PA3, STM32F7_PINMUX_FUNC_PA3_USART2_RX},
+#endif	/* CONFIG_UART_STM32_PORT_1 */
```
Comment (ydamigos) [inline boards/arm/stm32f723e_disco/pinmux.c]: CONFIG_UART_STM32_PORT_2?

**Review by ydamigos — CHANGES_REQUESTED** [review]
LGTM. Some comments need to be fixed.

**Comment by aurel32** [comment]
@ydamigos, thanks for the review and for finding some leftover in the comments. I have just pushed a new version.

**Review by erwango — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
docs LGTM, thanks.

**Pull request merged**